sphinx_search/extension.py

```diff
@@ -24,25 +26,65 @@ def _get_static_files(config):
     return ASSETS_FILES[file_type]
 
 
+def get_context(config):
+    """
+    Get context for templates.
+
+    This mainly returns the settings from the extension
+    that are needed in our JS code.
+    """
+    default_filter = config.rtd_sphinx_search_default_filter
+    filters = config.rtd_sphinx_search_filters
+    # When converting to JSON, the order of the keys is not guaranteed.
+    # So we pass a list of tuples to preserve the order.
+    filters = [(name, filter) for name, filter in filters.items()]
```

Comment:
I'd suppose you can workaround this with a `OrderedDict`, but I haven't tried it.

Reply:
Python 3.6 or 3.7 dicts keep the order, but the problem here is when converting that dictionary to a json object, json objects don't keep the order.